### ISSUE TITLE
Allow to specify binary request methods

### ISSUE THREAD

**Issue by khaliullov** [Miserlou/Zappa#696]
At the moment Zappa processes request body if request methods are in a following list: ```["POST", "PUT", "PATCH"]```.
```ENV['CONTENT_LENGTH']``` is not calculated for ```DELETE``` or other request types. It is impossible to change hard-coded list of methods with body. 

## Context
According RFC 2616 sec 9.2:
```OPTIONS``` request could include an entity-body.
Also, in the same RFC sec 4.3:
> The presence of a message-body in a request is signaled by the
>    inclusion of a Content-Length or Transfer-Encoding header field in
>    the request's message-headers. A message-body MUST NOT be included in
>    a request if the specification of the request method (section 5.1.1)
>    does not allow sending an entity-body in requests. A server SHOULD
>    read and forward a message-body on any request; if the request method
>    does not include defined semantics for an entity-body, then the
>    message-body SHOULD be ignored when handling the request.

So, Zappa should forward request body to the lambda handler.

## Expected Behavior
Request body should be detected by ```Content-Length``` or ```Transfer-Encoding``` header or just by ```len(event_info['body'])```. ```ENV['CONTENT_LENGTH']``` should be set and passed to the lambda handler.

## Actual Behavior
```ENV['wsgi.input']``` contains request body, but ```ENV['CONTENT_LENGTH']``` is not defined if request type is not in the hard-coded list: ```["POST", "PUT", "PATCH"]```

## Possible Fix
At the moment I think it is possible to workaround this issue by creating custom lambda_handler or in other monkey-patching way.
Possible solutions:
- Hard-code another request methods;
- Allow user to control the list of methods where entity-body could be present, for example via zappa_settings.json;
- Detect entity-body by ```Content-Length``` or ```Transfer-Encoding``` header or just by ```len(event_info['body'])```.

## Steps to Reproduce
For example, Django is not proccessing request body in case, if I want to implement following protocol:

```
DELETE /posts/ HTTP/1.1

[{"id": 1}, {"id": 2}]
```

## Your Environment
* Zappa version used: 0.33
* Operating System and Python version: Ubuntu 16.04
* Link to your project (optional):
* Your `zappa_settings.py`:

**Comment by Miserlou** [comment]
Is the problem here that they need to be specified, or that there are methods which simply aren't behaving as they should? I'd really like to make this an area that's zero-configuration if at all possible.

**Comment by Miserlou** [comment]
Found this: https://en.wikipedia.org/wiki/Hypertext_Transfer_Protocol#Summary_table

Is this acceptable?:

```
if binary_support and (method in ["POST", "PUT", "PATCH", "DELETE", "CONNECT", "OPTIONS"]):
```

**Comment by Miserlou** [comment]
Published this change to 0.37.2 - let me know if it's unacceptable and I'll reopen the ticket for further discussion.

**Comment by khaliullov** [comment]
@Miserlou, thank you for your attention to this issue!
As I said before - this way is acceptable (the easiest one), but `ENV['CONTENT_LENGTH']` is not calculated:
https://github.com/Miserlou/Zappa/blob/228f0bcaebbb3acd2d9fe4ce93112e3b2efea78a/zappa/wsgi.py#L86

if event_info has always 'body' attribute we can do it like this:

```
# https://en.wikipedia.org/wiki/Hypertext_Transfer_Protocol#Summary_table
if binary_support and event_info.get('isBase64Encoded', False):
    encoded_body = event_info['body']
    body = base64.b64decode(encoded_body)
else:
    body = event_info['body']
    body = str(body)

#...

# Input processing
if 'Content-Type' in headers:  # set header if we got it
    environ['CONTENT_TYPE'] = headers['Content-Type']
if len(body):  # request has body (` and method not in ["GET","HEAD", "TRACE"]`).. pass it to the handler
    environ['wsgi.input'] = StringIO(body)
    environ['CONTENT_LENGTH'] = str(len(body))
elif method in ["POST", "PUT", "PATCH"]:  # no body for methods with body
    environ['CONTENT_LENGTH'] = '0'
```
